app/exerlog/lib/main.dart

```diff
@@ -1,62 +1,108 @@
 // @dart = 2.9
-import 'package:exerlog/Bloc/authentication.dart';
 import 'package:exerlog/UI/calendar/view/calendar_page.dart';
+import 'package:exerlog/UI/global.dart';
+import 'package:exerlog/src/feature/authentication/controller/authentication_controller.dart';
+import 'package:exerlog/src/utils/logger/logger.dart';
+import 'package:exerlog/src/utils/logger/riverpod_logger.dart';
 import 'package:firebase_auth/firebase_auth.dart';
+import 'package:firebase_core/firebase_core.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
-import 'UI/login_screen/login_page.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
 
+import 'src/core/base/shared_preference/shared_preference_b.dart';
+import 'src/feature/authentication/view/landing_screen.dart';
+
+/// TODO: Remove global instances and use shared pref keys
+@Deprecated('Use `USER_UID` instead. Will be removed soon')
 String userID = '';
+@Deprecated('Should be removed')
 User the_user;
 
-void main() {
+void main() async {
   WidgetsFlutterBinding.ensureInitialized();
+
+  /// Device orientation locked to portrait up
   SystemChrome.setPreferredOrientations([
     DeviceOrientation.portraitUp,
   ]);
-  runApp(MyApp());
+
+  /// Initialize shared preference
+  try {
+    await initializeSharedPreference();
+  } catch (e, stackTrace) {
+    Log.error(e);

```

Comment:
Most of the time when we log an error we are gonna want to log the stack trace as well. So wouldn't it be better to include the stack trace in the log.error method as well? Something like :

`Log.error(e, stacktrace: stackTrace.toString());`